cvat-core/src/api-implementation.ts

```diff
@@ -529,19 +523,27 @@ export default function implementAPI(cvat: CVATCore): CVATCore {
 
         return mergedConflicts;
     });
-    implementationMixin(cvat.analytics.quality.settings.get, async (filter: QualitySettingsFilter) => {
-        checkFilter(filter, {
-            taskID: isInteger,
-        });
+    implementationMixin(
+        cvat.analytics.quality.settings.get,
+        async (filter: QualitySettingsFilter, aggregate?: boolean) => {
+            checkFilter(filter, {
+                taskID: isInteger,
+                projectID: isInteger,
+                parentType: isString,
+            });
 
-        const params = fieldsToSnakeCase(filter);
+            const params = fieldsToSnakeCase(filter);
 
-        const settings = await serverProxy.analytics.quality.settings.get(params);
-        const schema = await getServerAPISchema();
-        const descriptions = convertDescriptions(schema.components.schemas.QualitySettings.properties);
+            const settingsList = await serverProxy.analytics.quality.settings.get(params, aggregate);
+            if (settingsList) {
+                const schema = await getServerAPISchema();
+                const descriptions = convertDescriptions(schema.components.schemas.QualitySettings.properties);
 
-        return new QualitySettings({ ...settings, descriptions });
-    });
+                const settings = settingsList.map((setting) => new QualitySettings({ ...setting, descriptions }));
+                return settings;
+            }
+            return null;
```

Comment:
`null` is not declared as possible return type for this method
Additionally the endpoint returns list, so, I do not think it can be null.